<a href="https://colab.research.google.com/github/alexandralima1602/-C8280-./blob/main/actividad01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Descargamos el archivo llamado "data.tsv

In [18]:
!wget /content/data.tsv

/content/data.tsv: Scheme missing.


In [10]:
!cat data.tsv | column -t -s $'\t'

order_id  quantity  item_name                              choice_description                                                                                                                                                                                         item_price
1         1         Chips and Fresh Tomato Salsa           NULL                                                                                                                                                                                                       $2.39 
1         1         Izze                                   [Clementine]                                                                                                                                                                                               $3.39 
1         1         Nantucket Nectar                       [Apple]                                                                                                                           

**Mire en la parte superior e inferior de data.tsv entregado en esta actividad. Piensa en cómo se estructuran los datos. ¿Qué crees que significa cada columna? ¿Qué crees que significa cada fila?**

Mostrar la parte superior de los datos (las primeras filas)

In [2]:
!head data.tsv

order_id	quantity	item_name	choice_description	item_price
1	1	Chips and Fresh Tomato Salsa	NULL	$2.39 
1	1	Izze	[Clementine]	$3.39 
1	1	Nantucket Nectar	[Apple]	$3.39 
1	1	Chips and Tomatillo-Green Chili Salsa	NULL	$2.39 
2	2	Chicken Bowl	[Tomatillo-Red Chili Salsa (Hot), [Black Beans, Rice, Cheese, Sour Cream]]	$16.98 
3	1	Chicken Bowl	[Fresh Tomato Salsa (Mild), [Rice, Cheese, Sour Cream, Guacamole, Lettuce]]	$10.98 
3	1	Side of Chips	NULL	$1.69 
4	1	Steak Burrito	[Tomatillo Red Chili Salsa, [Fajita Vegetables, Black Beans, Pinto Beans, Cheese, Sour Cream, Guacamole, Lettuce]]	$11.75 
4	1	Steak Soft Tacos	[Tomatillo Green Chili Salsa, [Pinto Beans, Cheese, Sour Cream, Lettuce]]	$9.25 


 parte inferior de los datos (las últimas filas)

In [3]:
!tail data.tsv

1831	1	Chips	NULL	$2.15 
1831	1	Bottled Water	NULL	$1.50 
1832	1	Chicken Soft Tacos	[Fresh Tomato Salsa, [Rice, Cheese, Sour Cream]]	$8.75 
1832	1	Chips and Guacamole	NULL	$4.45 
1833	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Black Beans, Sour Cream, Cheese, Lettuce, Guacamole]]	$11.75 
1833	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Sour Cream, Cheese, Lettuce, Guacamole]]	$11.75 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Pinto Beans, Guacamole, Lettuce]]	$11.25 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Lettuce]]	$8.75 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Pinto Beans, Lettuce]]	$8.75 



RSPTA: Las columnas significan los parametros de toda la data como ID, cantidad, nombre del árticulo, elección_descripción y el precio del articulo, las filas significan los ejemplos de los parametros

**¿Cuántos pedidos parece haber?**

In [11]:
!awk -F'\t' '{ sum += $2 } END { print sum }' data.tsv


4972


Viendo la suma de la cantidad de pedidos llegamos a la conclusiòn de que hay 4972 pedidos

**¿cuantas lineas hay en este archivo?**

In [12]:
!wc -l data.tsv

4624 data.tsv


**¿Qué burrito es más popular, steako chicken?**

In [15]:
!grep -E "Steak Burrito" data.tsv | wc -l


368


In [16]:
!grep -E "Chicken Burrito" data.tsv | wc -l


553


El mas popular es Chiken Burrito por que tiene mayor cantidad de pedidos con 553

**¿Los burritos de pollo suelen tener frijoles negros o frijoles pintos?**

lo hice de 2 formas

**forma 1**

In [21]:
!awk -F'\t' '/Chicken Burrito/ && (/Black Beans/ ) { count++ } END { print count }' data.tsv


282


In [22]:
!awk -F'\t' '/Chicken Burrito/ && (/Pinto Beans/ ) { count++ } END { print count }' data.tsv

105


**forma 2**

In [24]:
!echo "Black Beans:"
!grep "Chicken Burrito" data.tsv | grep "Black Beans" | wc -l
!echo "Pinto Beans:"
!grep "Chicken Burrito" data.tsv | grep "Pinto Beans" | wc -l

Black Beans:
282
Pinto Beans:
105


Los burritos de pollo suelen tener frijoles negros como frijoles pintos

**Usa la línea de comandos para descubrir algo interesante sobre los datos dados.**

Para encontrar registros duplicados en función de una columna:

In [30]:
!cut -f1 data.tsv | sort | uniq -d


1
10
100
1000
1001
1002
1003
1005
1006
1007
1008
1009
101
1010
1011
1012
1013
1014
1015
1016
1017
1018
1019
102
1020
1021
1025
1026
1028
1029
103
1030
1031
1032
1033
1034
1035
1037
1038
1039
104
1040
1041
1042
1043
1044
1045
1046
1047
1049
1050
1051
1052
1053
1054
1055
1056
1057
1058
1059
106
1060
1062
1063
1064
1065
1066
1067
1068
1069
1070
1071
1072
1073
1074
1075
1076
1077
1078
1079
108
1080
1081
1082
1083
1084
1085
1086
1087
1088
1089
109
1090
1091
1092
1093
1094
1095
1096
1097
1098
1099
11
110
1101
1102
1103
1104
1105
1106
1107
1108
1109
111
1110
1111
1112
1113
1114
1115
1116
1117
1118
1119
112
1120
1121
1122
1123
1124
1125
1126
1127
1128
1129
113
1130
1131
1132
1133
1134
1135
1136
1137
1138
1139
114
1140
1141
1142
1143
1144
1145
1147
1148
1149
115
1150
1151
1152
1153
1154
1155
1156
1157
1158
1159
116
1160
1161
1162
1163
1164
1165
1166
1167
1168
1169
117
1170
1171
1172
1173
1176
1177
1178
1179
118
1180
1181
1182
1184
1185
1186
1187
1188
1189
119
1190
1191
1192
1193
1194
1195
1196


Para encontrar el ítem más pedido:

In [31]:
!cut -f3 data.tsv | sort | uniq -c | sort -nr | head -n 1

    726 Chicken Bowl
